Scripts/generate-test-config.sh

```diff
@@ -1,14 +1,269 @@
 #!/bin/bash
 
-DESTINATION=$1
+# vim: set ts=2 sw=2 sts=2 et:
 
-if [[ -z $DESTINATION ]] ; then
-  echo "Destination not specified" >&2
+##################################################
+# SCRIPT METADATA
+
+SCRIPT_NAME=$(basename $0)
```

Comment:
Bash supports the `readonly` keyword, which is good for global constant values, like these:
```
readonly SCRIPT_NAME="$(basename $0)"
readonly SYNOPSIS="$SCRIPT_NAME"
```
etc.

As an alternative, instead of forward-declaring these global vars, you could mimic the constructs of a general purpose programming language:

```
main() {
    # Bash (not shell) supports both final-assignment and function-scoped variables.
    readonly local program_name="$(dirname $0)"
    # ...
}

main $@
```

Reply:
Thanks for all the feedback on this--lots of new bash learnings for me!

Reply:
Thanks for the links to Shellguide & Shellcheck; I hadn't seen those before. The uppercase symbols are constants, which the checker had no way of knowing because I wasn't using the `readonly` keyword as @jamesonwilliams pointed out. I'll look at the recommendations from Shellcheck & address